mempool/src/rpc.rs

```diff
@@ -40,7 +55,31 @@ impl MempoolRpcServer for super::MempoolHandle {
         rpc::handle_result(self.call(move |this| this.contains_transaction(&tx_id)).await)
     }
 
-    async fn submit_transaction(&self, tx: HexEncoded<SignedTransaction>) -> rpc::Result<()> {
+    async fn contains_orphan_tx(&self, tx_id: Id<Transaction>) -> rpc::Result<bool> {
+        rpc::handle_result(self.call(move |this| this.contains_orphan_transaction(&tx_id)).await)
+    }
+
+    async fn get_transaction(&self, tx_id: Id<Transaction>) -> rpc::Result<Option<GetTxResponse>> {
+        let res: Option<_> = rpc::handle_result(
+            self.call(move |this| -> Result<_, crate::error::Error> {
+                let res = if let Some(tx) = this.transaction(&tx_id)? {
+                    Some((tx, TxStatus::InMempool))
+                } else {
+                    this.orphan_transaction(&tx_id)?.map(|tx| (tx, TxStatus::InOrphanPool))
+                };
+                Ok(res)
+            })
+            .await,
+        )?;
+
+        Ok(res.map(|(transaction, status)| GetTxResponse {
+            id: tx_id,
+            status,
+            transaction: HexEncoded::new(transaction),
+        }))
+    }
+
```

Comment:
Putting this here for the lack of a better location.

Can you add functional tests that prove that the orphan pool functions normally in a node?